### PR TITLE
docs: Replace py.test with pytest

### PR DESCRIPTION
**Pull request by shreyans800755**
Fixes https://github.com/coala/coala/issues/4189

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by thisch** [comment]
This looks good. (It has to be ack-ed by someone else, since I'm not a core-dev)

File: docs/Developers/Writing_Tests.rst [hunk 1]
```diff
@@ -30,28 +30,24 @@ automatically executed by running:
...
     Often you don't want to run all available tests. To run your
     specific one, type (in the coala root folder):
 
     .. code:: shell
 
-        $ py.test -k <your-test>
+        $ pytest -k <your-test>
 
     You can even give partial names or queries like "not MyCustomTest"
     to not run a specific test. More information can be got with
```
Comment (userzimmermann) [inline docs/Developers/Writing_Tests.rst]: Please change *can be got* with *is shown*. It just sounds better ;)

File: docs/Developers/Writing_Tests.rst [hunk 2]
```diff
@@ -187,7 +183,7 @@ as a kickstart for test-writing.
 
 Put the code under the desired folder inside ``tests``,
 modify it to let it test your stuff and run the test from
-the coala root folder ``py.test``.
+the coala root folder ``pytest``.
```
Comment (userzimmermann) [inline docs/Developers/Writing_Tests.rst]: Please add a little *using* before *pytest*. It also just sounds better ;)

File: docs/Developers/Executing_Tests.rst [hunk 3]
```diff
@@ -34,11 +34,7 @@ You can then execute our tests with
 
 ::
 
-    $ py.test
-
-.. note::
-    If ``py.test`` seems to give errors, try running ``python3 -m pytest``
-    instead.
```
Comment (SanketDG) [inline docs/Developers/Executing_Tests.rst]: I'd like to keep this. I saw @thisch's [comment](https://github.com/coala/coala/issues/4189#issuecomment-299944884) but this is immensely helpful for Windows users where setting a path is painstakingly difficult.

File: docs/Developers/Writing_Tests.rst (hunk 1 above)
Comment (SanketDG) [inline docs/Developers/Writing_Tests.rst]: separate commit

File: docs/Developers/Writing_Tests.rst (hunk 2 above)
Comment (SanketDG) [inline docs/Developers/Writing_Tests.rst]: separate commit

File: docs/Developers/Executing_Tests.rst (hunk 3 above)
Comment (shreyans800755) [inline docs/Developers/Executing_Tests.rst]: @thisch @userzimmermann 
What do you think ?

Comment (userzimmermann) [inline docs/Developers/Executing_Tests.rst]: @shreyans800755 Hmm... I think like @SanketDG: Let's keep that. But with `py.test` also changed to `pytest` of course ;P

**Comment by shreyans800755** [comment]
@userzimmermann 
Pushed both commits. Lemme know if anything else is missing in the commits.

File: docs/Developers/Writing_Tests.rst [hunk 4]
```diff
@@ -30,28 +30,24 @@ automatically executed by running:
...
+    $ pytest
 
 There's only one constraint:
 The name of the test file has to end with ``Test.py`` (for example
 ``MyCustomTest.py``, but not ``MyCustomTestSuite.py``).
 
 .. note::
-    If ``py.test`` seems to give errors, try running ``python3 -m pytest``
-    instead.
-
```
Comment (SanketDG) [inline docs/Developers/Writing_Tests.rst]: please keep this too

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4198/commits/8678fd1a1c08ad953da6bfd312de63ef5dd8050c

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/4198/commits/e19042762af4b8c57810823c8a4a7005bde98a97

**Comment by userzimmermann** [comment]
@shreyans800755 Done :) Thanks!

@SanketDG Good review comments! I would not have realized that it's a good idea to better keep those `python -m` hints ;)

**Comment by userzimmermann** [comment]
@rultor merge

**Pull request merged**